algorithms/sesotho.sbl

```diff
@@ -0,0 +1,109 @@
+/* stringescapes UTF-8 */
+
+/*
+  Sesotho stemmer for the Snowball project
+  ----------------------------------------
+  Author: Kamohelo Lebjane
+  Purpose: To reduce Sesotho words to their morphological stems.
+
+  Language notes:
+  Sesotho (Southern Sotho) is an agglutinative Bantu language.
+  Words often contain prefixes for noun classes and suffixes
+  for tense, aspect, or derivation.
+
+  An agglutinative language is a type of language that primarily forms words 
+  by stringing together morphemes (word parts)—each typically representing a single 
+  grammatical meaning—without significant modification to their forms (agglutinations). 
+  In such languages, affixes (prefixes, suffixes, infixes, or circumfixes) 
+  are added to a root word in a linear and systematic way, creating complex 
+  words that encode detailed grammatical information.
+
+  Examples:  
+    baruti      -> rut (root) 
+    moruti      -> rut (root)
+    rutile      -> rut (root)
+
+  The rules below remove common noun class prefixes
+  and common verb suffixes, keeping the main root form.
+*/
+
+/* --- Routine declarations --- */
+routines (
+  mark_regions
+  remove_noun_prefixes
+  remove_verb_suffixes
+  remove_nominal_suffixes
+)
+
+/* --- External declarations --- */
+externals ( stem )
+
+/* --- Groupings --- */
+groupings ( v )
+
+/* --- Character sets --- */
+define v 'aeiou'
+
+/* --- Integer for tracking position --- */
+integers ( pV )
+
+/* --- Mark vowel region --- */
+define mark_regions as (
+  $pV = limit
+  do (
+    gopast v  setmark pV
+  )
+)
+
+/* --- Remove noun class prefixes --- */
+define remove_noun_prefixes as (
+  [substring] among(
+    'mo'   'ba'   'me'   'le'
+    'ma'   'se'   'boi'   'li'
+  )
+  /* Only delete if there's a vowel after the cursor position */
+  (gopast v delete)
+)
+
+/* --- Remove verb suffixes (from end of word) --- */
+define remove_verb_suffixes as (
+  backwards (
+    setlimit tomark pV for (
+      [substring] among(
+        'ile'   /* perfect tense */
+        'isa'   /* causative */
+        'etse'  /* applicative */
+        'ela'   /* applicative */
+        'ang'   /* plural imperative */
+        'ong'    /* continuous/derived form */
+        'eng'
+        'wa'    /* passive */
+        'a'     /* infinitive marker */
+        (delete)
+      )
+    )
+  )
+)
+
+/* --- Remove nominal suffixes --- */
+define remove_nominal_suffixes as (
+  backwards (
+    setlimit tomark pV for (
+      [substring] among(
+        'nyana'   /* diminutive form */
+        'ana'     /* diminutive form */
+        'oa'
+        'i'
+        (delete)
+      )
+    )
+  )
+)
+
+/* --- MAIN STEMMER --- */
+define stem as (
+  do mark_regions
+  do remove_nominal_suffixes
+  do remove_verb_suffixes
+  do remove_noun_prefixes
+)
```

Comment:
Just a minor nit, but please include a newline character on the final line (github's red icon here means there isn't one).  It's better to include as some text processing tools can behave unhelpfully without one (and some editors will add automatically add one which can create noise in future PRs).